### ISSUE TITLE
Lucky's Tale stops displaying on headset at first underground area

### ISSUE THREAD

**Issue by LeoDavidson** [LibreVR/Revive#292]
Revive 1.0-preview f867beb (from today I think)
SteamVR beta 1477423729 (from today, adding ATW)
NVidia 375.63.
Lucky's Tale downloaded from Oculus store today.
Windows 10 AU 64-bit.

Initially, everything is working great and I can play the start of the first level. But when I get to the first underground bit, near the first checkpoint, he goes underground and I see the side-scrolling section, but, shortly after I move, the screen fades out to the SteamVR "void" environment.

The game is still running in the background, playing music and, from the mirrored window, still responding to the joypad and headset movements, but the graphics it is rendering aren't being displayed on the headset anymore.

SteamVR has a red message saying the game is unresponsive, even though it appears to be responding fine.

Repeated this twice in a row, with the same results each time.

Please let me know if I can provide any other information, or help to test anything. Happy to try fairly technical things if needed. (I'm a Windows developer, although I know very little about VR or game development.)

PS: Thank you for making Revive. Aside from this issue, it looks really slick and was very easy to get working.

**Comment by LibreVR** [comment]
Thanks for the report, I'll try to reproduce it. For now try rolling back to the previous release and see if that fixes it.

**Comment by LucidRealityVR** [comment]
I'd like to add that I am also getting this on Lucky's Tale. It happened after a certain amount of time (about 20 seconds) in the main menu. Then during the intro of the game. Same thing. Displays in window on monitor, only see SteamVR area in Vive, says "unresponsive" in SteamVR status

**Comment by LeoDavidson** [comment]
After going to the previous Revive version (installed 0.9.7 over the top of 1.0-preview), everything seemed to work fine. I played through all the Chapter 1 levels without issue (other than one or two places in the levels where the framerate dropped but remained playable).

FWIW, I did this with interleaved reprojection off in SteamVR, leaving only asynchronous reprojection on, but I don't think the reprojection settings matter as I tried the same while still on 1.0-preview and I still got stuck in the SteamVR environment while the game was only visible on the 2D monitor.

While still on 1.0-preview, I also tried various combinations of reprojection settings and also changing the "Status Check" interval under SteamVR's Developer options page, figuring it may be related to SteamVR deciding the game was no longer responding, but none of the settings seemed to make a difference. (I changed the settings without restarting anything, as it seems like just closing the settings dialog and re-opening it always resets to "Normal (1 sec)".)

In some cases changing things seemed to make the problem worse, with it not always working past the intro screens or "push A" menu. Or maybe they had no effect and it's slightly random. My guess is that it happens when or shortly after the game loads a new area, but it's just a guess.

**Comment by LibreVR** [comment]
I think this is a deadlock related to multi-threading since I haven't put much effort into providing the same thread-safety guarantees as Oculus.

**Comment by LibreVR** [comment]
This has been fixed in my local development version. Unfortunately the only fix I was able to find was to not apply the optimization that was added in the 1.0 preview for Unity games.

Due to the way Unity uses different threads to access the tracking information and submit the VR frames it triggers a deadlock with OpenVR. I'll try to fix this properly in a future version, but it may require changes to be made on the side of OpenVR.
